bin/all_sky_search/pycbc_template_kde_calc

```diff
@@ -0,0 +1,120 @@
+    maxFOM = FOM[(optbw, optalpha)]
+    return optbw, optalpha
+
+#calling template data and constructing template_kde_hdf file over template params
+template_file = h5py.File(args.template_file, 'r')
+mass1_n = template_file['mass1'][:]
+mass2_n = template_file['mass2'][:]
+spin1z_n = template_file['spin1z'][:]
+spin2z_n = template_file['spin2z'][:]
+chi_eff_n = convert.chi_eff(mass1_n, mass2_n, spin1z_n, spin2z_n)
+mchirp_n = convert.mchirp_from_mass1_mass2(mass1_n, mass2_n)
+eta_n = convert.eta_from_mass1_mass2(mass1_n, mass2_n)
+
+if args.make_template_kde:
+    sample_n = numpy.vstack((numpy.log(mchirp_n), eta_n, chi_eff_n)).T
+    grid_pts_n = sample_n
+    optbw_n, optalpha_n = optimizedparam(sample_n, nfold=2)
+    kde_data_n = kde_awkde(sample_n, grid_pts_n, alp=optalpha_n, gl_bandwidth=optbw_n)
+
+    hdf = h5py.File(args.output_file, 'w')
+    hdf.create_dataset('kde_template', data=kde_data_n)
+    hdf.attrs['bank_kde'] = 'kde_template'
+    hdf.create_dataset('mass1', data=mass1_n)
+    hdf.create_dataset('mass2', data=mass2_n)
+    hdf.create_dataset('spin1z', data=spin1z_n)
+    hdf.create_dataset('spin2z', data=spin2z_n)
+    hdf.create_dataset('mchirp', data=mchirp_n)
+    hdf.create_dataset('eta', data=eta_n)
+    hdf.create_dataset('chi_eff', data=chi_eff_n)
+    hdf.create_dataset('grid_pts', data=grid_pts_n)
+    hdf.close()
+
+#calling signal data and constructing signal_kde_hdf file over template params
+if args.make_signal_kde:
+    data_signal = numpy.genfromtxt(args.signal_file, dtype = float, delimiter=',', names = True)
+    mass2 = data_signal['mass2']
+    mass2_threshold = 5
+    N_original = len(mass2)
+    if mass2_threshold:
+        idx = mass2 > mass2_threshold
+        mass2 = mass2[idx]
+        logging.info('%i triggers out of %i with MASS2 > %s' %
+                         (len(mass2), N_original, str(mass2_threshold)))
+    else:
+        idx = numpy.full(N_original, True)
+    mass2_s = mass2
+    mass1_s = data_signal['mass1'][idx]
+    q_s = data_signal['q'][idx]
+    eta_s = data_signal['eta'][idx]
+    chi_eff_s  = data_signal['chi_eff'][idx]
+    mchirp_s = data_signal['mchirp'][idx]
+
+    sample_s = numpy.vstack((numpy.log(mchirp_s), eta_s, chi_eff_s)).T
+    grid_pts_s = numpy.array((numpy.log(mchirp_n), eta_n, chi_eff_n)).T
+    optbw_s, optalpha_s = optimizedparam(sample_s, nfold=2)
+    kde_data_s = kde_awkde(sample_s, grid_pts_s, alp=optalpha_s, gl_bandwidth=optbw_s)
+
+    hdf = h5py.File(args.output_file, 'w')
+    hdf.create_dataset('kde_sig', data=kde_data_s)
+    hdf.attrs['signal_kde'] = 'kde_sig'
+    hdf.create_dataset('mass1', data=mass1_s)
+    hdf.create_dataset('mass2', data=mass2_s)
+    hdf.create_dataset('mchirp', data=mchirp_s)
+    hdf.create_dataset('eta', data=eta_s)
+    hdf.create_dataset('chi_eff', data=chi_eff_s)
+    hdf.create_dataset('grid_pts', data=grid_pts_s)
+    hdf.close()
+
+if args.make_signal_kde and args.make_template_kde:
```

Comment:
This input checking should be done immediately after the parse_args statement. It is not a good idea to do a bunch of calculations that may take some time and _then_ terminate the code with an error message and no output.  If there is a problem with the options or inputs the user shuld be told as early as possible.